src/plugins/integrations/PostgreIntegration.ts

```diff
@@ -131,7 +110,40 @@ class PostgreIntegration implements Integration {
 
                     span._initialized();
 
-                    return originalRes;
+                    let originalCallback: any;
+                    let callbackIndex = -1;
+
+                    for (let i = 1; i < newArgs.length; i++) {
+                        if (typeof newArgs[i] === 'function') {
+                            originalCallback = newArgs[i];
+                            callbackIndex = i;
+                            break;
+                        }
+                    }
+
+                    if (callbackIndex >= 0) {
+                        const wrappedCallback = (err: any, res: any) => {
+                            if (err) {
+                                span.setErrorTag(err);
+                            }
+                            span.closeWithCallback(me, originalCallback, [err, res]);
+                        };
+                        newArgs[callbackIndex] = wrappedCallback;
+                    }
+
+                    const result = query.apply(this, newArgs);
+
+                    if (result && typeof result.then === 'function') {
+                        result.then(function (value: any) {
+                            span.closeWithCallback(me , originalCallback, [null, value]);
```

Comment:
As far as I see, when it returns promise, there will be no callback. So do we still need close with callback here? Can't we just close span here? Is it valid case to use callback even it expects promise?